src/main/java/com/mlaf/hu/decisionagent/behavior/RegisterBehavior.java

```diff
@@ -0,0 +1,31 @@
+package com.mlaf.hu.decisionagent.behavior;
+
+import com.mlaf.hu.decisionagent.DecisionAgent;
+import com.mlaf.hu.models.InstructionSet;
+import jade.core.behaviours.CyclicBehaviour;
+import jade.lang.acl.ACLMessage;
+import jade.lang.acl.MessageTemplate;
+
+public class RegisterBehavior extends CyclicBehaviour{
+    private final DecisionAgent DA;
+
+    public RegisterBehavior(DecisionAgent da) {
+        this.DA = da;
+    }
+
+    @Override
+    public void action() {
+        ACLMessage message = myAgent.receive(MessageTemplate.MatchPerformative(ACLMessage.SUBSCRIBE));
+        if (message != null) {
```

Comment:
Deze if kan beter worden gereversed voor leesbaarheid
if (message == null) {
    return;
}
Andere code...

Reply:
Ben ik het niet helemaal mee eens. @rogierpijpers  jij bent java boy. Wat vind jij? Ik vind het zelf op deze manier strak.

Reply:
Ik ben t met Auke eens, maar da's smaak. Hoe minder indents hoe beter als t aan mij ligt. Voor mij het liefste zelf zo (zonder brackets als t om een one liner gaat):
if (message == null)
    return;

Andere code.

Maareh, jou code en t is niet fout zoals t is, dus kies verstandig.